partialWaveFit/fitResult.cc

```diff
@@ -243,103 +262,188 @@ fitResult::fill(const unsigned int              nmbEvents,               // numb
+                const complexMatrix&            normIntegral,            // normalization integral matrix
+                const complexMatrix&            acceptedNormIntegral,    // normalization integral matrix with acceptance
+                const vector<double>&           phaseSpaceIntegral,      // normalization integral over full phase space without acceptance
+                const bool                      converged,
+                const bool                      hasHessian)
+{
+	fill(nmbEvents, normNmbEvents, multibinBoundaries, logLikelihood, rank, prodAmps, prodAmpNames,
+	     &fitParCovMatrix, fitParCovMatrixIndices, &normIntegral, &acceptedNormIntegral, &phaseSpaceIntegral,
+	     converged, hasHessian);
+}
+
 
 void
-fitResult::fill(const fitResult& result)
+fitResult::fill(const fitResult& result,
+                const TMatrixT<double>*         fitParCovMatrix,         // covariance matrix of fit parameters
+                const vector<pair<int, int> >*  fitParCovMatrixIndices,  // indices of fit parameters for real and imaginary part in covariance matrix matrix
+                const complexMatrix*            normIntegral,            // normalization integral matrix
+                const complexMatrix*            acceptedNormIntegral,    // normalization integral matrix with acceptance
+                const vector<double>*           phaseSpaceIntegral)      // normalization integral over full phase space without acceptance
 {
-	_nmbEvents              = result.nmbEvents();
-	_normNmbEvents          = result.normNmbEvents();
-	_multibinBoundaries     = result.multibinBoundaries();
-	_logLikelihood          = result.logLikelihood();
-	_rank                   = result.rank();
-	_prodAmps               = result.prodAmps();
-	_prodAmpNames           = result.prodAmpNames();
-	_waveNames              = result.waveNames();
-	_covMatrixValid         = result.covMatrixValid();
-
-	const TMatrixT<double>& fitParCovMatrix = result.fitParCovMatrix();
-	_fitParCovMatrix.ResizeTo(fitParCovMatrix.GetNrows(), fitParCovMatrix.GetNcols());
-	_fitParCovMatrix        = fitParCovMatrix;
-
-	_fitParCovMatrixIndices = result.fitParCovIndices();
-	_normIntegral           = result.normIntegralMatrix();
-	_acceptedNormIntegral   = result.acceptedNormIntegralMatrix();
-	_normIntIndexMap        = result.normIntIndexMap();
-	_phaseSpaceIntegral     = result.phaseSpaceIntegralVector();
-	_converged              = result.converged();
-	_hasHessian             = result.hasHessian();
+	if (fitParCovMatrix == nullptr)
+		fitParCovMatrix = (not result.covMatrixValid()) ? nullptr : &result.fitParCovMatrix();
+	if (fitParCovMatrixIndices == nullptr)
+		fitParCovMatrixIndices = &result.fitParCovIndices();
+	if (normIntegral == nullptr)
+		normIntegral = (result.normIntegralMatrix().nRows() == 0 and result.normIntegralMatrix().nCols() == 0) ? nullptr : &result.normIntegralMatrix();
```

Comment:
`and` -> `or`

Reply:
I think it is correct his way, because you want to interpret the matrix only as empty if rows==0 and cols ==0.
Everything else should raise a warning, as it does this way.